### PR TITLE
chore: Bump Snaps dependencies

### LINKED ISSUE #2639: Allow Button to wrap more elements
Allow `Button` to wrap more elements such as `Image`, `Icon` etc.

### LINKED ISSUE #2640: Add Selector component
"Rich" dropdown with more detailed elements than just text.

Details TBD

### LINKED ISSUE #2658: Expose a color prop for `Text`
We should consider exposing a color prop on `Text`, allowing devs to choose between a subset of color options.

For this initial version we should add the following colors: default, alternative, muted, error, success, warning

### PR DESCRIPTION
**Pull request by FrederikBolding**
## **Description**
Bump snaps packages to latest and handle breaking changes.

Summary of changes in the snaps deps:
- Add `Selector` component
- Add `Icon` component
- Add `color` prop to `Text` component
- `Button` children are now allowed to be `Image` and `Icon`

Closes https://github.com/MetaMask/snaps/issues/2640
Closes https://github.com/MetaMask/MetaMask-planning/issues/1615
Closes https://github.com/MetaMask/snaps/issues/2658
Closes https://github.com/MetaMask/snaps/issues/2639

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26675?quickstart=1)

### CODE REVIEW

**Review by ritave — APPROVED** [review]

File: ui/pages/permissions-connect/permissions-connect.component.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ const APPROVE_TIMEOUT = MILLISECOND * 1200;
 
 function getDefaultSelectedAccounts(currentAddress, permissionsRequest) {
   const permission =
-    permissionsRequest.permissions?.[RestrictedMethods.eth_accounts];
+    permissionsRequest?.permissions?.[RestrictedMethods.eth_accounts];
```
Comment (david0xd) [inline ui/pages/permissions-connect/permissions-connect.component.js:29]: Why is this change needed?

**Review by david0xd — APPROVED** [review]

File: ui/pages/permissions-connect/permissions-connect.component.js (hunk 1 above)
Comment (FrederikBolding) [reply ui/pages/permissions-connect/permissions-connect.component.js:29]: I saw some crashes in E2E due to this, I think it's a race condition we hadn't considered.

Unrelated to the other changes in this PR, but seemed worth fixing.

**Comment by bowensanders** [comment]
QA Passes ✅
